resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/RestClientDelegateBean.java

```diff
@@ -110,10 +135,103 @@ public Object create(CreationalContext<Object> creationalContext) {
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException("Could not find hostname verifier class" + verifier, e);
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException("Failed to instantiate hostname verifier class. Make sure it has a public, no-argument constructor", e);
+        } catch (ClassCastException e) {
+            throw new RuntimeException("The provided hostname verifier " + verifier + " is not an instance of HostnameVerifier", e);
+        }
+    }
+
+    private void registerKeyStore(String keyStorePath, RestClientBuilder builder) {
+        Optional<String> keyStorePassword = getOptionalProperty(KEY_STORE_PASSWORD, String.class);
+        Optional<String> keyStoreType = getOptionalProperty(KEY_STORE_TYPE, String.class);
+
+        try {
+            KeyStore keyStore = KeyStore.getInstance(keyStoreType.orElse("JKS"));
+            String password = keyStorePassword.orElseThrow(() -> new IllegalArgumentException("No password provided for keystore"));
+
+            try (InputStream input = locateStream(keyStorePath)) {
+                keyStore.load(input, password.toCharArray());
+            } catch (IOException | CertificateException | NoSuchAlgorithmException e) {
+                throw new IllegalArgumentException("Failed to initialize trust store from classpath resource " + keyStorePath, e);
+            }
+
+            builder.keyStore(keyStore, password);
+        } catch (KeyStoreException e) {
+            throw new IllegalArgumentException("Failed to initialize trust store from " + keyStorePath, e);
+        }
+    }
+
+    private void registerTrustStore(String trustStorePath, RestClientBuilder builder) {
+        Optional<String> maybeTrustStorePassword = getOptionalProperty(TRUST_STORE_PASSWORD, String.class);
+        Optional<String> maybeTrustStoreType = getOptionalProperty(TRUST_STORE_TYPE, String.class);
+
+        try {
+            KeyStore trustStore = KeyStore.getInstance(maybeTrustStoreType.orElse("JKS"));
+            String password = maybeTrustStorePassword.orElseThrow(() -> new IllegalArgumentException("No password provided for truststore"));
+
+            try (InputStream input = locateStream(trustStorePath)) {
+                trustStore.load(input, password.toCharArray());
+            } catch (IOException | CertificateException | NoSuchAlgorithmException e) {
+                throw new IllegalArgumentException("Failed to initialize trust store from classpath resource " + trustStorePath, e);
+            }
+
+            builder.trustStore(trustStore);
+        } catch (KeyStoreException e) {
+            throw new IllegalArgumentException("Failed to initialize trust store from " + trustStorePath, e);
+        }
+    }
+
+    private InputStream locateStream(String path) throws FileNotFoundException {
+        if (path.startsWith("classpath:")) {
+            if (path.startsWith("classpath:")) {
```

Comment:
Duplicated check? ;-)

Reply:
fixed, sorry, I missed that when cleaning up